internal/common/predicate.go

```diff
@@ -12,3 +14,16 @@ func IsControlledBy(object client.Object, owner *metav1.OwnerReference) bool {
 		controller.APIVersion == owner.APIVersion &&
 		controller.Kind == owner.Kind
 }
+
+// FetchCurrentNamespaceName returns namespace name by using information stored as file
+// Returns default Openshift Builds namespace on error
+// Refer: https://kubernetes.io/docs/tasks/run-application/access-api-from-pod/#without-using-a-proxy
+func FetchCurrentNamespaceName() string {
+	namespace, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		CurrentNamespaceName = OpenShiftBuilNamespaceName
+	} else {
+		CurrentNamespaceName = string(namespace)
+	}
+	return CurrentNamespaceName
```

Comment:
Avoid global state/variables here. Suggestion also includes the typo fix.

```suggestion
	if err != nil {
		return OpenShiftBuildNamespaceName
	}
	return string(namespace)
```